File: Derangements/Derangements.py
 Derangements.py                                                                                                                                      
   1 def derangements(n):
   2     if n == 0:
   3         return 1
   4     if n == 1:
   5         return 0
   6     return (n-1) * (derangements(n-1) + derangements(n-2))
   7 n = int(input('Enter n: '))
   8 print(derangements(n))
   9 
  10 
~                                                                                                                                                     
~                                                                                                                                                     
~                                                         

File: Derangements/Derangements.java
import java.io.*;
import java.util.*;
class Main {
   
    // Function to count derangements 
    static int countDer(int n) {
        // Base case
          if(n == 1 || n == 2) {
            return n-1;
        }
       
        // Variable for storing prev values
        int a = 0;
          int b = 1;
       
        // manner using above recursive formula
        for (int i = 3; i <= n; ++i) {
            int cur = (i-1)*(a+b);
            a = b;
              b = cur;
        }
             
        // Return result for n
        return b;
    }
       
    // Driver Code
    public static void main (String[] args) 
    {
        int n ;
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        System.out.println("Count of Dearrangements is " + 
                            countDer(n));
       
    }
   
  // Code contributed by skagnihotri
}
